portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,71 @@
 
 package com.google.sps.servlets;
 
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private static final String COMMENT_NAME = "name";
+  private static final String COMMENT_TEXT = "comment";
+  private static final String COMMENT_TIMESTAMP = "timestamp";
+
+  private static List<Comment> comments = new ArrayList<>();
+
+  @Override
+  public void init() {
+    // Retrieve stored comments
+    Query query = new Query("Comment").addSort(COMMENT_TIMESTAMP, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()) {
```

Comment:
This could be more concise with Java8 streams. 

Consider using results.asList([params]).forEach(entity -> comments.add(Comment.fromEntity(entity));